yb-voyager/src/srcdb/postgres.go

```diff
@@ -507,3 +509,31 @@ func (pg *PostgreSQL) ClearMigrationState(migrationUUID uuid.UUID, exportDir str
 	log.Infof("ClearMigrationState not implemented yet for PostgreSQL")
 	return nil
 }
+
+func (pg *PostgreSQL) GetReplicationConnection() (*pgconn.PgConn, error) {
+	return pgconn.Connect(context.Background(), pg.getConnectionUri()+"&replication=database")
+}
+
+func (pg *PostgreSQL) CreateLogicalReplicationSlot(conn *pgconn.PgConn, migrationUUID uuid.UUID, dropIfAlreadyExists bool) (pglogrepl.CreateReplicationSlotResult, error) {
+	replicationSlotName := fmt.Sprintf("voyager_%s", strings.Replace(migrationUUID.String(), "-", "_", -1))
+	res, err := pglogrepl.CreateReplicationSlot(context.Background(), conn, replicationSlotName, "pgoutput",
+		pglogrepl.CreateReplicationSlotOptions{Mode: pglogrepl.LogicalReplication})
+
+	if err != nil {
+		if strings.Contains(err.Error(), "already exists") && dropIfAlreadyExists {
+			// drop and recreate
+			err = pglogrepl.DropReplicationSlot(context.Background(), conn, replicationSlotName, pglogrepl.DropReplicationSlotOptions{})
+			if err != nil {
+				return pglogrepl.CreateReplicationSlotResult{}, fmt.Errorf("failed to delete existing replication slot: %v", err)
```

Comment:
If you change `CreateLogicalReplicationSlot()` to return a **pointer** to `pglogrepl.CreateReplicationSlotResult`, then you can simply return `nil` in case of errors.